# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/shistory/ChildSHistory.h
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc docshell/shistory/ChildSHistory.h
--- a/docshell/shistory/ChildSHistory.h	Wed Apr 28 11:09:38 2021 +0000
+++ b/docshell/shistory/ChildSHistory.h	Wed Apr 28 11:26:49 2021 +0000
@@ -67,13 +67,14 @@
    * backwards.
    */
   bool CanGo(int32_t aOffset);
-  void Go(int32_t aOffset, bool aRequireUserInteraction, ErrorResult& aRv);
+  void Go(int32_t aOffset, bool aRequireUserInteraction, bool aUserActivation,
+          ErrorResult& aRv);
   void AsyncGo(int32_t aOffset, bool aRequireUserInteraction,
-               CallerType aCallerType, ErrorResult& aRv);
+               bool aUserActivation, CallerType aCallerType, ErrorResult& aRv);
 
   // aIndex is the new index, and aOffset is the offset between new and current.
   void GotoIndex(int32_t aIndex, int32_t aOffset, bool aRequireUserInteraction,
-                 ErrorResult& aRv);
+                 bool aUserActivation, ErrorResult& aRv);
 
   void RemovePendingHistoryNavigations();
 
@@ -101,16 +102,19 @@
         public mozilla::LinkedListElement<PendingAsyncHistoryNavigation> {
    public:
     PendingAsyncHistoryNavigation(ChildSHistory* aHistory, int32_t aOffset,
-                                  bool aRequireUserInteraction)
+                                  bool aRequireUserInteraction,
+                                  bool aUserActivation)
         : Runnable("PendingAsyncHistoryNavigation"),
           mHistory(aHistory),
           mRequireUserInteraction(aRequireUserInteraction),
+          mUserActivation(aUserActivation),
           mOffset(aOffset) {}
 
     NS_IMETHOD Run() override {
       if (isInList()) {
         remove();
-        mHistory->Go(mOffset, mRequireUserInteraction, IgnoreErrors());
+        mHistory->Go(mOffset, mRequireUserInteraction, mUserActivation,
+                     IgnoreErrors());
       }
       return NS_OK;
     }
@@ -118,6 +122,7 @@
    private:
     RefPtr<ChildSHistory> mHistory;
     bool mRequireUserInteraction;
+    bool mUserActivation;
     int32_t mOffset;
   };
 